src/lib.rs

```diff
@@ -598,6 +598,29 @@ impl<T> Receiver<T> {
         }
     }
 
+    /// Returns the current state of the channel.
+    #[inline]
+    pub fn state(&self) -> ChannelState {
+        // SAFETY: the existence of the `self` parameter serves as a certificate that the receiver
+        // is still alive, meaning that even if the sender was dropped then it would have observed
+        // the fact that we're still alive and left the responsibility of deallocating the
+        // channel to us, so `self.channel` is valid
+        let channel = unsafe { self.channel_ptr.as_ref() };
+
+        match channel.state.load(Acquire) {
+            DISCONNECTED => ChannelState::Disconnected,
+            EMPTY => ChannelState::Empty,
+            MESSAGE => ChannelState::Message,
+            RECEIVING => ChannelState::Receiving,
```

Comment:
I don't think it makes sense for the `Receiver` to be able to tell that it's receiving. Do you have a valid use case for this? Except for in some weird async cases, I don't think a receiver can ever end up in this state, because the only time it is receiving is while it is executing inside one of the receiving methods.